lib/ancestry/class_methods.rb

```diff
@@ -7,7 +7,7 @@ def to_node(object)
       if object.is_a?(ancestry_base_class)
         object
       else
-        unscoped_where { |scope| scope.find(object.try(primary_key) || object) }
+        unscoped_where { |scope| scope.find(object.try(ancestry_identifier_column) || object) }
```

Comment:
here, primary_key (`:id`), vs ancestry_identifier_column (`:id`) -- I'll keep them the same for now, but want to set this to find other issues

Reply:
and find inherently is a `find_by(:id => object.id)`
So this should read `find_by(ancestry_identifier_column => object.read_attribute(ancestry_identifier_column))`

As I'm thinking more about this, this getting unwieldy very quickly.
ods are pretty low.

By the way, this method could be defined as `def to_node(id_or_object)`. The `try` is saying "if this is an `Object`, then lets fetch the `id`, otherwise lets assume it is an `Integer` `id`", which works because ruby dropped the `Object#id` maybe 15 years ago.'